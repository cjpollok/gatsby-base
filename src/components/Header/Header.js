import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'gatsby'

import styles from './header.module.scss'

const Header = ({ siteTitle }) => (
  <header className={styles.container}>
    <div className={styles.menu}>
      <h1 className={styles.homeLink}>
        <Link to="/">{siteTitle}</Link>
      </h1>
    </div>
  </header>
)

Header.propTypes = {
  siteTitle: PropTypes.string.isRequired,
}

export default Header
