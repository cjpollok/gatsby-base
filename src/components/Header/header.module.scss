@import '../../styles/colors.scss';

.container {
  background: $header;
}

.menu {
  margin: 0 auto;
  max-width: 960px;
  padding: 1.5rem 1rem;
}

.homeLink {
  margin: 0;

  a {
    color: $white;
    text-decoration: none;
  }
}
