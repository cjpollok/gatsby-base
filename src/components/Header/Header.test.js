import React from 'react'
import { render } from 'enzyme'

import Header from './Header'

describe('Header Component', () => {
  it('renders correctly', () => {
    const wrapper = render(<Header siteTitle="Site Title" />)
    expect(wrapper).toMatchSnapshot()
  })
})
