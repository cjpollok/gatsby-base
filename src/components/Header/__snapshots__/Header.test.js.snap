// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<header
  class="container"
>
  <div
    class="menu"
  >
    <h1
      class="homeLink"
    >
      <a
        href="/"
      >
        Site Title
      </a>
    </h1>
  </div>
</header>
`;
